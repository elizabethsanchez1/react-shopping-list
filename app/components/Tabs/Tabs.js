import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text, View, StyleSheet, Dimensions } from 'react-native'
import theme from "../../styles/theme.style";
import { SceneMap, TabBar, TabView } from "react-native-tab-view";


const styles = StyleSheet.create({
  label: {
    color: '#fff',
    fontSize: theme.FONT_SIZE_LARGE,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
  },
  shadow: {
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 3},
    shadowOpacity: 0.8,
    shadowRadius: 2,
    elevation: 1,
    backgroundColor: theme.SECONDARY_BACKGROUND,
    // changes to black background
    // backgroundColor: theme.PRIMARY_BACKGROUND,
  },
  tabHeader: {
    // marginTop: 30,
  },
});


/*
* TODO
* Needs base height set inside of here just in case its not being set inside the component
* */



class Tabs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
      routes: props.routes,
    };

    console.log('props passed in: ', props);
  }

  _renderTabBar = props => (
    <TabBar
      {...props}
      indicatorStyle={{
        backgroundColor: theme.ACTIVE_TAB_COLOR
      }}
      //renderIcon={this._renderIcon}
      labelStyle={styles.label}
      // tabStyle={{ height: 200 }}
      // style={
      //   (props.background === 'dark')
      //     ? { ...styles.shadow, backgroundColor: theme.SECONDARY_BACKGROUND }
      //     : { ...styles.shadow, backgroundColor: theme.PRIMARY_BACKGROUND }
      // }
      style={{ ...styles.shadow, ...this.props.tabBarStyling }}
    />
  );

  _renderScene = SceneMap(this.props.subViews);

  render() {
    console.log('props in tabs: ', this.props);

    return(
      <TabView
        style={{...styles.tabHeader, ...this.props.tabHeaderStyling}}
        navigationState={this.state}
        renderScene={this._renderScene}
        onIndexChange={index => this.setState({ index })}
        initialLayout={{
          width: Dimensions.get('window').width,
          height: Dimensions.get('window').height
        }}
        renderTabBar={this._renderTabBar}
      />
    )
  }
}


Tabs.propTypes = {
  tabBarStyling: PropTypes.object,
  tabHeaderStyling: PropTypes.object,
};

export default Tabs;
