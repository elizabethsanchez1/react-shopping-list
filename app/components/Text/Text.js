import React from 'react';
import PropTypes from 'prop-types';
import { Text as BaseText, StyleSheet } from 'react-native';
import theme from "../../../app/styles/theme.style";


const styles = StyleSheet.create({
  header: {
    fontSize: theme.FONT_SIZE_HEADERBAR,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },
  standardText: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },
  small: {
    fontSize: theme.FONT_SIZE_SMALL,
  },
  medium: {
    fontSize: theme.FONT_SIZE_MEDIUM,
  },
  large: {
    fontSize: theme.FONT_SIZE_LARGE,
  },
  lightFontWeight: {
    fontWeight: theme.FONT_WEIGHT_LIGHT,
  },
  mediumFontWeight: {
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
  },
  boldFontWeight: {
    fontWeight: theme.FONT_WEIGHT_BOLD,
  }
});


// TODO: Font weight does not work for some reason???

const Text = ({ children, size, weight, textStyling }) => {
  let styleObject = { ...styles.standardText  };

  if ( size === 'small' ) {
    styleObject = { ...styleObject, ...styles.small };
  }

  if ( size === 'medium' ) {
    styleObject = { ...styleObject, ...styles.medium };
  }

  if ( size === 'large' ) {
    styleObject = { ...styleObject, ...styles.large };
  }

  if ( size === 'header' ) {
    styleObject = { ...styleObject, ...styles.header };
  }

  // if ( weight === 'light' ) {
  //   styleObject = { ...styleObject, ...styles.lightFontWeight }
  // }
  //
  // if ( weight === 'medium' ) {
  //   styleObject = { ...styleObject, ...styles.mediumFontWeight }
  // }
  //
  // if ( weight === 'bold' ) {
  //   styleObject = { ...styleObject, ...styles.boldFontWeight }
  // }

  styleObject = { ...styleObject, ...textStyling };

  return (
    <BaseText
      style={ styleObject }
    >
      { children }
    </BaseText>
  )
};

Text.propTypes = {
  children: PropTypes.any,
  size: PropTypes.string,
  textStyling: PropTypes.object,
};

export default Text;
