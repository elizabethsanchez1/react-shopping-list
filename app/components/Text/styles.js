import { create } from 'react-native-platform-stylesheet';
import theme from '../../styles/theme.style';

export default create({
  header: {
    fontSize: theme.FONT_SIZE_HEADERBAR,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },

  // Text specific styles

  standardText: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },
});
