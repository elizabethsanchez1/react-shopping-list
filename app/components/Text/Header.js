import PropTypes from 'prop-types';
import React from 'react';
import { Text } from 'react-native';
import styles from './styles';

const Header = (props) => (
  <Text
    style={[styles.header, props.style]}
  >
    {props.children}
  </Text>
);

Header.propTypes = {
  children: PropTypes.any,
  style: PropTypes.any,
};

export default Header;
