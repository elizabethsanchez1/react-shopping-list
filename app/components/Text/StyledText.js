import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';
import styles from './styles';

const StyledText = (props) => (
  <Text
    style={[styles.standardText, props.style]}
  >
    {props.children}
  </Text>
);


StyledText.propTypes = {
  children: PropTypes.any,
  style: PropTypes.any,
};

export default StyledText;
