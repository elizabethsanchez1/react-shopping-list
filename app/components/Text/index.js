/* eslint-disable import/prefer-default-export */
import Header from './Header';
import StyledText from './StyledText';
import Text from './Text';

export { Header, StyledText, Text };
