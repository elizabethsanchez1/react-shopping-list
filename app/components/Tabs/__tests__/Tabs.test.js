import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { View } from "react-native";
import { Text } from "../../Text";
import Tabs from "../Tabs";


describe( 'Tabs Component', () => {

  const firstView = () => (
    <View><Text>First tab content</Text></View>
  );

  const secondView = () => (
    <View><Text>Second tab content</Text></View>
  );


  it( 'Tabs base visually looks the same', () => {
    const tree = renderer.create(
      <View style={{width: '100%', height: 100}}>
        <Tabs
          routes={[
            { key: 'first', title: 'FIRST TAB' },
            { key: 'second', title: 'SECOND TAB' },
          ]}
          subViews={{
            first: firstView,
            second: secondView,
          }}
        />
      </View>
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );

} );




