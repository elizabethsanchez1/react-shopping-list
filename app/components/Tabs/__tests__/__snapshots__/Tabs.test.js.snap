// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component Tabs base visually looks the same 1`] = `
<View
  style={
    Object {
      "height": 100,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#47474D",
          "elevation": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "zIndex": 1,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#72DEFF",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 375,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "overflow": "scroll",
          }
        }
      >
        <RCTScrollView
          alwaysBounceHorizontal={false}
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "nowrap",
              },
              Object {
                "flex": 1,
              },
            ]
          }
          horizontal={true}
          keyboardShouldPersistTaps="handled"
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          overScrollMode="never"
          scrollEventThrottle={1}
          scrollsToTop={false}
          showsHorizontalScrollIndicator={false}
        >
          <View>
            <View
              accessibilityLabel="FIRST TAB"
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontFamily": "RobotoCondensed-Regular",
                        "fontSize": 18,
                        "margin": 8,
                      }
                    }
                  >
                    FIRST TAB
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessibilityLabel="SECOND TAB"
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "justifyContent": "center",
                      "opacity": 0.7,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontFamily": "RobotoCondensed-Regular",
                        "fontSize": 18,
                        "margin": 8,
                      }
                    }
                  >
                    SECOND TAB
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        alwaysBounceHorizontal={false}
        automaticallyAdjustContentInsets={false}
        bounces={false}
        contentContainerStyle={null}
        contentOffset={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        directionalLockEnabled={true}
        horizontal={true}
        keyboardDismissMode="on-drag"
        keyboardShouldPersistTaps="always"
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        overScrollMode="never"
        pagingEnabled={true}
        scrollEventThrottle={1}
        scrollsToTop={false}
        showsHorizontalScrollIndicator={false}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View>
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Object {
                "overflow": "hidden",
                "width": 750,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#FFF",
                    "fontFamily": "RobotoCondensed-Regular",
                    "fontSize": 16,
                    "fontWeight": "600",
                  }
                }
              >
                First tab content
              </Text>
            </View>
          </View>
          <View
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            style={
              Object {
                "overflow": "hidden",
                "width": 750,
              }
            }
          >
            <View />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
